src/api/app/controllers/status/reports_controller.rb

```diff
@@ -6,6 +6,8 @@ class Status::ReportsController < ApplicationController
   # GET /status_reports/published/:project_name/:repository_name/reports/:uuid
   # GET /status_reports/build/:project_name/:repository_name/:arch/reports/:uuid
   def show
+    return raise ActiveRecord::RecordNotFound, 'Empty status report' if @status_report.nil?
```

Comment:
In `set_status_report`, `@checkable.status_reports.find_by!(uuid: params[:report_uuid])` will already raise `ActiveRecord::RecordNotFound` if a status report isn't found. We can do the same for the other line (`@checkable.status_reports.first`) by using `first!` instead of `first`.